core/domain/auth_services_test.py

```diff
@@ -231,3 +235,63 @@ def test_delete_association_when_it_is_present(self) -> None:
     def test_delete_association_when_it_is_missing_does_not_raise(self) -> None:
         # Should not raise.
         auth_services.delete_external_auth_associations('does_not_exist')
+
+    def test_auth_session_established_or_destoryed(self) -> None:
+        def mock_establish_auth_session(
+            _: webapp2.Request,
+            __: webapp2.Response
+        ) -> None:
+            raise Exception('Is Established')
```

Comment:
Semantically, raising an exception is weird here because there's no error happening. Can you instead declare a list in the test function, and append to that list in the individual functions, then assert the contents of that list? The messages you write can be similar to 'establish_auth_session method called', etc.

Reply:
done!